common/headers/versionChecker.go

```diff
@@ -52,39 +56,43 @@ var (
 		ClientNameCLI:     "<2.0.0",
 		ClientNameServer:  "<2.0.0",
 	}
+	SupportedClientFeatures = map[string]map[string]string{
+		FeatureFollowsNextRunID: {
+			ClientNameGoSDK:   ">=1.10.0",
+			ClientNameJavaSDK: ">=1.4.0",
+			ClientNameServer:  ">=1.13.0",
+		},
+	}
 )
 
 type (
 	// VersionChecker is used to check client/server compatibility and client's capabilities
 	VersionChecker interface {
 		ClientSupported(ctx context.Context, enableClientVersionCheck bool) error
+		ClientSupportsFeature(ctx context.Context, feature string) bool
```

Comment:
I tried to combine these into one function, but it got pretty messy since `ClientSupported` has lots of logic to make nice error messages, and opposite semantics for unknown client ids, compared to `ClientSupportsFeature`.

Reply:
@dnr we're adding a new method as a handshake between client and server, I'm not sure if it could be used here but maybe we should send the capabilities from the client instead of version checking?
https://github.com/temporalio/temporal/issues/2190

Reply:
This change does use capabilities sent by the client instead of versions. I don't think we followed up on the client side, though. We should start sending those capabilities.

It's the opposite direction from #2190. I think we need both.